projects/pastanaga-angular/src/lib/side-nav/side-nav.component.ts

```diff
@@ -0,0 +1,96 @@
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import {
+    Component,
+    ChangeDetectionStrategy,
+    Input,
+    ContentChildren,
+    QueryList,
+    AfterContentInit,
+    ViewChild,
+    ChangeDetectorRef,
+    ElementRef,
+    Renderer2,
+    OnInit,
+} from '@angular/core';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { markForCheck } from '../common';
+import { PastanagaService } from '../pastanaga.service';
+import { SideNavItemComponent } from './side-nav-item.component';
+
+@Component({
+    selector: 'pa-side-nav',
+    templateUrl: './side-nav.component.html',
+    styleUrls: ['./side-nav.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SideNavComponent implements OnInit, AfterContentInit {
+    @Input()
+    get inverted(): boolean {
+        return this._inverted;
+    }
+    set inverted(value: boolean) {
+        this._inverted = coerceBooleanProperty(value);
+    }
+    @Input()
+    get visible(): boolean {
+        return !!this._visible;
+    }
+    set visible(value: boolean) {
+        this._visible = coerceBooleanProperty(value);
+    }
+    @Input() toggle?: Subject<boolean>;
```

Comment:
why is this `toggle` an Input?
I feel like toggling the navbar on mobile/desktop is all internal, isn't it? you can toggle it by clicking on the close button or on the overlay, both being inside the navbar, no?